packages/beacon-node/src/api/impl/beacon/blocks/index.ts

```diff
@@ -418,9 +418,11 @@ export function getBeaconBlockApi({
       if (!blobSidecars) {
         throw Error(`blobSidecars not found in db for slot=${block.message.slot} root=${toHexString(blockRoot)}`);
       }
+
       return {
         executionOptimistic,
-        data: blobSidecars,
+        // Return all sidecars if no indices are specified, only the requested ones (identified by `indices`) otherwise
+        data: indices.length == 0 ? blobSidecars : blobSidecars.filter(({index}) => indices.includes(index)),
```

Comment:
can't this just be simplified to, this makes to code also clear enough, you can keep the comment but I don't think it's needed

```suggestion
        data: indices ? blobSidecars.filter(({index}) => indices.includes(index)) : blobSidecars,
```